### PR TITLE
Add PRIMA derivative-free optim lib

### PR DESCRIPTION
**Pull request by jschueller**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jschueller** [comment]
should be good to go @conda-forge/staged-recipes

File: recipes/prima/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{% set version = "0.7.1" %}
+
+package:
+  name: prima
+  version: {{ version }}
+
+source:
+  fn: prima-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/prima/meta.yaml:8]: You probably don't need this but it doesn't hurt. No strong opinion on removing or keeping it. I'll leave to you.

File: recipes/prima/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  run_exports:
+    - {{ pin_subpackage('prima', max_pin='x.x') }}
+
+requirements:
+  build:
+    - cmake
+    - ninja
+    - {{ compiler("c") }}               # [unix]
+    - {{ compiler("fortran") }}         # [unix]
+    # flang does not support F2018, use gfortran
```
Comment (ocefpaf) [inline recipes/prima/meta.yaml:25]: Pinging @isuruf just out of curiosity. You probably already know about this though :grimacing:

File: recipes/prima/meta.yaml (hunk 1 above)
Comment (jschueller) [reply recipes/prima/meta.yaml]: yes, lets remove it if useless

File: recipes/prima/meta.yaml (hunk 2 above)
Comment (jschueller) [reply recipes/prima/meta.yaml:24]: in particular, flang does not support procedure pointers that we use in the C interface (i'm not sure its actually part of f2018 though)

**Comment by jschueller** [comment]
thanks !

**Pull request merged**